bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/BigtableIOHBase.java

```diff
@@ -0,0 +1,204 @@
+    private static final long serialVersionUID = 1L;
+
+    private static final DoFn<Row, Result> ROW_TO_RESULT_TRANSFORM = new DoFn<Row, Result>() {
+      private static final long serialVersionUID = 1L;
+      @ProcessElement
+      public void processElement(ProcessContext c) throws Exception {
+        c.output(Adapters.ROW_ADAPTER.adaptResponse(c.element()));
+      }
+    };
+
+    private final SerializableConfiguration serializableConfiguration;
+    private final String tableId;
+    private final SerializableScan scan = new SerializableScan(new Scan());
+
+    private Read(String projectId, String instanceId, String tableId) {
+      Preconditions.checkNotNull(projectId, "Please configure a projectId.");
+      Preconditions.checkNotNull(instanceId, "Please configure a instanceId.");
+      Preconditions.checkNotNull(tableId, "Please configure a tableId.");
+      this.serializableConfiguration =
+          new SerializableConfiguration(BigtableConfiguration.configure(projectId, instanceId));
+      this.tableId = tableId;
+    }
+
+    /**
+     * Add an extended configuration option.  See {@link BigtableOptionsFactory} for more details.
+     */
+    public Read withConfiguration(String key, String value) {
+      serializableConfiguration.get().set(key, value);
+      return this;
+    }
+
+    /**
+     * Attach a {@link RowFilter} and potentially a {@link ByteKeyRange} to a {@link Read}.
+     */
+    public Read withScan(Scan scan) {
+      this.scan.set(scan);
+      return this;
+    }
+
+    @Override
+    public void populateDisplayData(DisplayData.Builder builder) {
+        super.populateDisplayData(builder);
+        builder.add(DisplayData.item("configuration",
+                serializableConfiguration.get().toString()));
+        builder.add(DisplayData.item("tableId", tableId));
+        builder.addIfNotNull(DisplayData.item("scan", scan.get().toString()));
+    }
+
+    @Override
+    public PCollection<Result> expand(PBegin input) {
+      RowFilter filter = Adapters.SCAN_ADAPTER.adapt(scan.get(), new DefaultReadHooks()).getFilter();
+      byte[] startRow = scan.get().getStartRow();
+      byte[] stopRow = scan.get().getStopRow();
+      ByteKeyRange keyRange = ByteKeyRange.ALL_KEYS;
+      if (startRow.length > 0){
+        keyRange   = keyRange.withStartKey(ByteKey.copyFrom(startRow));
+      }
+      if (stopRow.length > 0){
+        keyRange = keyRange.withEndKey(ByteKey.copyFrom(stopRow));
+      }
+
+      BigtableOptions options =
+          BigtableOptionsFactory.fromConfiguration(serializableConfiguration.get());
+      BigtableIO.Read bigtableRead =
+          BigtableIO.read()
+              .withBigtableOptions(options)
+              .withKeyRange(keyRange)
+              .withRowFilter(filter)
+              .withTableId(tableId);
+      return input.getPipeline()
+          .apply("BigtableIO.Read", bigtableRead)
+          .apply("HBase Result transformer", ParDo.of(ROW_TO_RESULT_TRANSFORM));
```

Comment:
Would be good to keep the naming consistent. Either CloudBigtableIO.transform or Bigtable read